src/components/StatTable/StatTable.jsx

```diff
@@ -0,0 +1,179 @@
+import { useState, useEffect } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { NPOBackend } from '../../utils/auth_utils';
+import { Select } from '@chakra-ui/react';
+import './StatTable.css';
+
+const StatTable = ({ season, allSeasons }) => {
+  const [stats, setStats] = useState([]);
+  // const currSeason = season.split(' ')[0];
+  // const currYear = season.split(' ')[1];
+  // const [selectedSeason, setSelectedSeason] = useState(currSeason.toLowerCase());
+  // const [selectedYear, setSelectedYear] = useState(currYear);
+  // const [allSeasons, setAllSeasons] = useState([]);
+  const [selectedSeason, setSelectedSeason] = useState('');
+  // const uniqueSeasons = Array.from(new Set(allSeasons.map(season => season.split(' ')[0])));
+  // const uniqueYears = Array.from(new Set(allSeasons.map(seasonYear => seasonYear.split(' ')[1])));
+  // const toast = useToast();
+
+  console.log(season, allSeasons);
+
+  // useEffect(() => {
+  //   const renderTable = async () => {
+  //     const { data } = await NPOBackend.get('/published-schedule/all-seasons');
+  //     if (data.indexOf(curSeason) == -1) {
+  //       data.unshift(curSeason);
+  //     }
+
+  //     setSelectedSeason(curSeason);
+
+  //     const seasonOrder = ['Fall', 'Summer', 'Spring'];
+  //     data.sort((a, b) => {
+  //       // Compare years first
+  //       if (a.split(' ')[1] !== b.split(' ')[1]) {
+  //         return b.split(' ')[1] - a.split(' ')[1];
+  //       } else {
+  //         return seasonOrder.indexOf(a.split(' ')[0]) - seasonOrder.indexOf(b.split(' ')[0]);
+  //       }
+  //     });
+
+  //     setAllSeasons(data);
+
+  //   };
+  //   renderTable();
+  // }, [curSeason]);
+
+  useEffect(() => {
+    const fetchStats = async () => {
+      try {
+        if (!selectedSeason) {
+          setSelectedSeason(season);
+        }
+        const curSeason = selectedSeason.split(' ')[0];
+        const curYear = selectedSeason.split(' ')[1];
+        const response = await NPOBackend.get(`/published-schedule/stats?season=${curSeason.toLowerCase()}&year=${curYear}`);
+        const data = response.data;
+        setStats(data);
+      } catch (error) {
+        console.error('Error fetching statistics:', error);
+      }
+    };
+
+    fetchStats();
+  }, [selectedSeason]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'season'. Either include it or remove the dependency array. If 'setSelectedSeason' needs the current value of 'season', you can also switch to useReducer instead of useState and read 'season' in the reducer.